### PR TITLE
Removes pre-C99 build and header cruft

### PR DESCRIPTION
**Pull request by derobins**
* Assumes ANSI C headers exist
    * Assumes stdbool.h, stdint.h, and inttypes.h are present
    * Assumes the C++ compiler can handle stdint.h
    * Removes all work-arounds for missing functionality, especially stdbool.h

### CODE REVIEW

File: config/cmake_ext_mod/ConfigureChecks.cmake [hunk 1]
```diff
@@ -237,11 +233,6 @@ macro (HDF_FUNCTION_TEST OTHER_TEST)
   endif ()
 endmacro ()
 
-#-----------------------------------------------------------------------------
-# Check for these functions before the time headers are checked
-#-----------------------------------------------------------------------------
-HDF_FUNCTION_TEST (STDC_HEADERS)
```
Comment (derobins) [inline config/cmake_ext_mod/ConfigureChecks.cmake:243]: The time headers are greatly simplified on modern POSIX systems. Those will be addressed in HDF5 in a later PR.

File: config/cmake_ext_mod/ConfigureChecks.cmake [hunk 2]
```diff
@@ -416,13 +407,9 @@ HDF_CHECK_TYPE_SIZE (time_t          ${HDF_PREFIX}_SIZEOF_TIME_T)
...
-# _Bool type support
-if (HAVE_STDBOOL_H)
-  set (CMAKE_EXTRA_INCLUDE_FILES stdbool.h)
-  HDF_CHECK_TYPE_SIZE (bool         ${HDF_PREFIX}_SIZEOF_BOOL)
-else ()
-  HDF_CHECK_TYPE_SIZE (_Bool        ${HDF_PREFIX}_SIZEOF_BOOL)
-endif ()
+# Size of bool
+set (CMAKE_EXTRA_INCLUDE_FILES stdbool.h)
+HDF_CHECK_TYPE_SIZE (_Bool        ${HDF_PREFIX}_SIZEOF_BOOL)
```
Comment (derobins) [inline config/cmake_ext_mod/ConfigureChecks.cmake:412]: This is currently unused in the library, but I'm leaving it in place for parity with other datatypes.

File: config/cmake_ext_mod/HDFTests.c [hunk 3]
```diff
@@ -88,14 +88,6 @@ int main(void)
 }
 #endif
 
-#ifdef STDC_HEADERS
-#include <stdlib.h>
-#include <stdarg.h>
-#include <string.h>
-#include <float.h>
-int main() { return 0; }
-#endif /* STDC_HEADERS */
```
Comment (derobins) [inline config/cmake_ext_mod/HDFTests.c:97]: This is the CMake "test" for certain C headers being present. It needs to be completely removed.

**Comment by derobins** [comment]
If this is merged to 1.12.x or earlier as it will trigger binary compatibility issues due to the missing symbols, though this is probably a lower-order concern.

**Review by qkoziol — APPROVED** [review]

**Comment by qkoziol** [comment]
"Good riddance!"  :-)

**Review by lrknox — APPROVED** [review]

**Review by byrnHDF — APPROVED** [review]

**Pull request merged**